go/private/rules/binary.bzl

```diff
@@ -81,6 +88,34 @@ def _go_binary_impl(ctx):
         ),
     ]
 
+    if go.cgo_tools and go.mode.link in (LINKMODE_C_ARCHIVE, LINKMODE_C_SHARED):
+        cgo_exports = ctx.actions.declare_file("%s.h" % name)
+        concat_args = ctx.actions.args()
```

Comment:
I think it's a bug that we currently concatenate header files (#2131). The `go` command doesn't do this. We should just generate one header from the `main` package.

We probably need to gather other indirectly included headers from cgo packages and cdeps though.

Reply:
this is a very good point regarding transitive cc dependencies. i'll take a look.

Reply:
i think iterating on cdeps and merging ccinfo should be enough, i'll add a test case.